cpp/oneapi/dal/backend/primitives/reduction/reduction_rm_cw_naive_dpc.cpp

```diff
@@ -0,0 +1,157 @@
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*******************************************************************************/
+
+#include "oneapi/dal/backend/primitives/reduction/reduction_rm_cw_dpc.hpp"
+
+#include "oneapi/dal/backend/primitives/common.hpp"
+
+namespace oneapi::dal::backend::primitives {
+
+template <typename Float, typename BinaryOp, typename UnaryOp>
+class kernel_reduction_rm_cw_naive {
+    using inp_t = const Float*;
+    using out_t = Float*;
+
+public:
+    kernel_reduction_rm_cw_naive(inp_t const input,
+                                 out_t const output,
+                                 std::int64_t height,
+                                 std::int32_t lstride,
+                                 const BinaryOp& binary,
+                                 const UnaryOp& unary)
+            : input_{ input },
+              output_{ output },
+              unary_{ unary },
+              binary_{ binary },
+              height_{ height },
+              lstride_{ lstride } {}
+
+    void operator()(sycl::nd_item<2> it) const {
+        using sycl::ONEAPI::reduce;
+        // Common for whole WG
+        const auto col_idx = it.get_global_id(0);
+        const auto loc_idx = it.get_local_id(1);
+        const auto range = it.get_global_range(1);
+        // Exclusive for EU
+        Float acc = binary_.init_value;
+        for (std::int64_t i = loc_idx; i < height_; i += range) {
+            inp_t const inp_row = input_ + lstride_ * i;
+            acc = binary_.native(acc, unary_(inp_row[col_idx]));
+        }
+        // WG reduction
+        auto grp = it.get_group();
+        output_[col_idx] = reduce(grp, acc, binary_.native);
+    }
+
+private:
+    inp_t const input_;
+    out_t const output_;
+    const UnaryOp unary_;
+    const BinaryOp binary_;
+    const std::int64_t height_;
+    const std::int32_t lstride_;
+};
+
+template <typename Float, typename BinaryOp, typename UnaryOp>
+reduction_rm_cw_naive<Float, BinaryOp, UnaryOp>::reduction_rm_cw_naive(sycl::queue& q,
+                                                                       std::int64_t wg)
+        : q_(q),
+          wg_(wg) {
+    ONEDAL_ASSERT(0 < wg_ && wg_ <= device_max_wg_size(q_));
+}
+
+template <typename Float, typename BinaryOp, typename UnaryOp>
+reduction_rm_cw_naive<Float, BinaryOp, UnaryOp>::reduction_rm_cw_naive(sycl::queue& q)
+        : reduction_rm_cw_naive(q, device_max_wg_size(q)) {}
+
+template <typename Float, typename BinaryOp, typename UnaryOp>
+sycl::event reduction_rm_cw_naive<Float, BinaryOp, UnaryOp>::operator()(
+    const Float* input,
+    Float* output,
```

Comment:
Inconsistency between function declaration and definition. Function is declared here `reduction/reduction_rm_cw_dpc.hpp:35` with `inp_t` and `out_t`, but in definition `const Float*` and `Float*` used. 

If type alias is introduced, use it consistently in all functions, especially between declaration and definition, or remove it at all.

Reply:
Fixed